core/src/main/scala/org/apache/pekko/persistence/r2dbc/state/javadsl/R2dbcDurableStateStore.scala

```diff
@@ -44,6 +45,9 @@ class R2dbcDurableStateStore[A](scalaStore: ScalaR2dbcDurableStateStore[A])(impl
   override def deleteObject(persistenceId: String): CompletionStage[Done] =
     scalaStore.deleteObject(persistenceId).toJava
 
+  override def deleteObject(persistenceId: String, revision: Long): CompletionStage[Done] =
```

Comment:
I had to add this implementation because the originally this project used an older version of akka-projection which didn't have this method in their interface.

On the other hand the the latest akka BSL version of akka-persistence (and hence pekko-persistence) has this method which has to be implemented.

I copied the implementation from `pekko-persistence-jdbc` which seems to ignore the `revision` parameter https://github.com/apache/incubator-pekko-persistence-jdbc/blob/db5733e60fffb99d3d822d3d60199ec2f0890278/core/src/main/scala/org/apache/pekko/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala#L117-L118

Reply:
this is fine - we've got similar changes in the other pekko-persistence-* projects - we should consider supporting the revision flag but we could do this later

Reply:
Okay in that case then PR is ready to be reviewed